### PR TITLE
Run debug tests only in build and test workflow

### PR DESCRIPTION
**Pull request by zsmb13**
### Description

Release tests are no different from debug tests, there's no need to run them separately. We already have another job that checks if release builds are successful.

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

**Review by bychkovdmitry — APPROVED** [review]

**Review by bychkovdmitry — APPROVED** [review]

File: stream-chat-android-ui-common/src/test/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListItemLiveDataBenchmark.kt [hunk 1]
```diff
@@ -65,7 +65,7 @@ internal class MessageListItemLiveDataBenchmark {
             }
         }
         println("changing typing information 100 times on a message list with ${items.size} items took $duration milliseconds")
-        Truth.assertThat(duration).isLessThan(20)
```
Comment (JcMinarro) [inline stream-chat-android-ui-common/src/test/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListItemLiveDataBenchmark.kt:68]: Why this change?

**Review by JcMinarro — APPROVED** [review]

File: stream-chat-android-ui-common/src/test/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListItemLiveDataBenchmark.kt (hunk 1 above)
Comment (zsmb13) [reply stream-chat-android-ui-common/src/test/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListItemLiveDataBenchmark.kt:68]: This test was flaky, sometimes it failed because the operations it measured took 21ms instead of 20ms, this makes it more tolerant
